components/mosart/src/riverroute/RtmMod.F90

```diff
@@ -1932,21 +1970,32 @@ subroutine Rtmini(rtm_active,flood_active)
 !#endif
 
 
- do nt = 1,nt_rtm
-      do nr = rtmCTL%begr,rtmCTL%endr
-      
-       call UpdateState_hillslope(nr,nt)
-       call UpdateState_subnetwork(nr,nt)   
-                                          
-        rtmCTL%volr(nr,nt) = (TRunoff%wt(nr,nt) + TRunoff%wr(nr,nt) + TRunoff%wh(nr,nt)*rtmCTL%area(nr)*TUnit%frac(nr))  ! times "TUnit%frac( nr )" or not ?
-        if (inundflag .and. nt == 1) then  
-           rtmCTL%volr(nr,nt) = rtmCTL%volr(nr,nt) + TRunoff%wf_ini( nr )
-        else        
-           call UpdateState_mainchannel(nr,nt)
-        endif
+    do nt = 1,nt_rtm
+        do nr = rtmCTL%begr,rtmCTL%endr
         
-      enddo
- enddo
+           call UpdateState_hillslope(nr,nt)
+           call UpdateState_subnetwork(nr,nt)   
+
+            !rtmCTL%volr(nr,nt) = (TRunoff%wt(nr,nt) + TRunoff%wr(nr,nt) + TRunoff%wh(nr,nt)*rtmCTL%area(nr)*TUnit%frac(nr))  ! times "TUnit%frac( nr )" or not ?
+            ! TODO: check if it is consistent when inundflag and sediflag turned on together
+            if(sediflag) then
+               call UpdateState_mainchannel(nr,nt)
+               rtmCTL%volr(nr,nt) = (TRunoff%wt(nr,nt) + TRunoff%wr(nr,nt) + &
+                                     TRunoff%wh(nr,nt)*rtmCTL%area(nr)*TUnit%frac(nr) + &
+                                     TRunoff%wt_al(nr,nt) + TRunoff%wr_al(nr,nt))
+            else
+               rtmCTL%volr(nr,nt) = (TRunoff%wt(nr,nt) + TRunoff%wr(nr,nt) + &
+                                     TRunoff%wh(nr,nt)*rtmCTL%area(nr)*TUnit%frac(nr))
+            end if
+
+            if (inundflag .and. nt == 1) then  
+               rtmCTL%volr(nr,nt) = rtmCTL%volr(nr,nt) + TRunoff%wf_ini( nr )
+            else
+               call UpdateState_mainchannel(nr,nt)
```

Comment:
When sediflag = true but inundflag = false, UpdateState_mainchannel will be called twice. Is it necessary?

Reply:
In addressing your comments above, this redundancy issue is also addressed.